db/judge/BaseJudge.js

```diff
@@ -59,11 +66,26 @@ class BaseJudge {
     return aux
   }
 
+  fail (msg) {
+    throw new IncorrectSolutionError(msg)
+  }
+
   /*
    * Returns false, if solution is not a valid solution
    * or integer, designating the grade of the solution
    */
-  judge (solution) {
+  async performJudgement (...params) {
+    try {
+      return await this.judge(...params)
+    }
+    catch (e) {
+      if (e instanceof IncorrectSolutionError) {
```

Comment:
There is a problem with portability. Assignments do not throw an `IncorrectSolutionError` for the moment.

Reply:
This change is backwards compatible.